Atomic/pull.py

```diff
@@ -1,16 +1,33 @@
-from .syscontainers import SystemContainers
+try:
+    from . import Atomic
+except ImportError:
+    from atomic import Atomic  # pylint: disable=relative-import
+from .util import skopeo_copy, get_atomic_config, skopeo_inspect, decompose
+
+ATOMIC_CONFIG = get_atomic_config()
+
 
 def cli(subparser):
     # atomic pull
+    backend = ATOMIC_CONFIG.get('default_storage', "ostree")
     pullp = subparser.add_parser("pull", help=_("pull latest image from a repository"),
                                  epilog="pull the latest specified image from a repository.")
     pullp.set_defaults(_class=Pull, func='pull_image')
-    pullp.add_argument("--storage", dest="backend", help=_("Specify the storage."))
+    pullp.add_argument("--storage", dest="backend", default=backend,
+                       help=_("Specify the storage. Default is currently '%s'.  You can"
+                              "change the default by editing /etc/atomic.conf and changing"
+                              "the 'default_storage' field." % backend))
     pullp.add_argument("image", help=_("image id"))
 
-class Pull(SystemContainers):
-    def __init__(self):
-        super(Pull, self).__init__()
 
+class Pull(Atomic):
     def pull_image(self):
-        super(Pull, self).pull_image()
+        if self.args.backend == 'ostree':
+            return self.syscontainers.pull_image()
```

Comment:
Syscontainers.pull_image at this point has some redundant code as it also checks for the backend.  We should replace Syscontainers.pull_image() to use directly `self._pull_image_to_ostree(self._get_ostree_repo(), self.args.image, True)` and drop all the different handlers logic there